File: functions/alsa-lib/variants/c/0_alaw_to_s16_5.c
#include <stdint.h>

static int alaw_to_s16(unsigned char a_val)
{
	int t = a_val ^ 0x55;
	t = (t & 0x0F) | ((t & 0x70) << 4) | 0x80;
	int seg = (a_val & 0x70) >> 4;
	if(seg) t += 0x100;
	t <<= seg ? seg : 0;
	return (a_val & 0x80) ? t : -t;
}


int main(){
    alaw_to_s16('a');
}

File: functions/alsa-lib/variants/go/0_alaw_to_s16_5.go
package main

func alawToS16(aVal byte) int32 {
	aVal ^= 0x55
	t := int32(aVal & 0x7f)
	if t < 16 {
		t = (t << 4) + 8
	} else {
		seg := (t >> 4) & 0x07
		t = 0x108 + ((t & 0x0f) << 4)
		t <<= seg - 1
	}
	return t * int32(2*aVal&0x80 - 1)
}


func main(){
    alawToS16('a')
}